File: Go/Two Pointers/Container-With-Most-Water.go
package main

import "math"

func main() {

}

func maxArea(height []int) int {
	result, left, right := 0, 0, len(height)-1

	for left < right {
		container := int(math.Min(float64(height[left]), float64(height[right])) * (float64(right) - float64(left)))

		result = int(math.Max(float64(container), float64(result)))

		if height[left] < height[right] {
			left++
		} else {
			right--
		}
	}
	return result
}

File: Javascript/Two Pointers/Container-With-Most-Water.js
/**
 * @param {number[]} height
 * @return {number}
 */
var maxArea = function (height) {
  let [left, right] = [0, height.length - 1];
  let result = 0;

  while (left < right) {
    const container = Math.min(height[left], height[right]) * (right - left);
    result = Math.max(result, container);

    if (height[left] < height[right]) {
      left++;
    } else {
      right--;
    }
  }

  return result;
};
